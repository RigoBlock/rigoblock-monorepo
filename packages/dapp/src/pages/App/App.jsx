import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import Counter from '../../components/Counter'
import './App.scss'

class App extends Component {
  render() {
    return (
      <div className="app-wrapper">
        <h1>{this.props.title}</h1>
        <Link to="/vault" className="link">
          Vault
        </Link>
      </div>
    )
  }
}

App.propTypes = {
  title: PropTypes.string.isRequired
}

export default App
