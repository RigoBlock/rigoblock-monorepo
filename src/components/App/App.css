.app-wrapper {
  text-align: center;
}
