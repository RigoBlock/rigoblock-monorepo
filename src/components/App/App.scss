.app-wrapper {
  text-align: center;
  color: white;
}
