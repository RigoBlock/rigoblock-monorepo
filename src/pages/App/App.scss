.app-wrapper {
  text-align: center;
  color: white;
}

.link {
  font-size: 1.5em;
  text-decoration: none;
  background-color: #EEEEEE;
  color: #333333;
  padding: 2px 6px 2px 6px;
}
