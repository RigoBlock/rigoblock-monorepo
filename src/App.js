import React, { Component } from 'react'
import './App.css'

class App extends Component {
  render() {
    return (
      <div className="App-wrapper">
        <h1>RigoBlock</h1>
      </div>
    )
  }
}

export default App
