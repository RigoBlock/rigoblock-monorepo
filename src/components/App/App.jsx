import React, { Component } from 'react'
import { Provider } from 'react-redux'
import configureStore from '../../store'
import PropTypes from 'prop-types'
import Counter from '../Counter'
import './App.scss'

const store = configureStore()
class App extends Component {
  render() {
    return (
      <div className="app-wrapper">
        <h1>RigoBlock</h1>
        <Provider store={store}>
          <Counter />
        </Provider>
      </div>
    )
  }
}

App.propTypes = {
  children: PropTypes.object
}

export default App
