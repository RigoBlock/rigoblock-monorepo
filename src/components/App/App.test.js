import React from 'react'
import ReactDOM from 'react-dom'
import { render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import App from './App.jsx'
import toJson from 'enzyme-to-json'

describe('App component', () => {
  const createComponentWithTitle = (title = 'RigoBlock') => {
    return (
      <MemoryRouter>
        <App title={title} />
      </MemoryRouter>
    )
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(createComponentWithTitle(), div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders correctly', () => {
    expect(toJson(render(createComponentWithTitle()))).toMatchSnapshot()
  })
})
