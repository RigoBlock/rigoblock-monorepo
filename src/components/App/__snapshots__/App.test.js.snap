// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  class="app-wrapper"
>
  <h1>
    RigoBlock
  </h1>
  <button>
    <a
      href="/counter"
    >
      Counter
    </a>
  </button>
</div>
`;
